adlfs/tests/test_spec.py

```diff
@@ -2045,3 +2048,15 @@ def test_open_file_x(storage: azure.storage.blob.BlobServiceClient, tmpdir):
         with fs.open("data/afile", "xb") as f:
             pass
     assert fs.cat_file("data/afile") == b"data"
+
+
+def test_user_agent(storage: azure.storage.blob.BlobServiceClient, mocker):
```

Comment:
It would be great if we could expand + parameterize this test case to assert user agent being set for:
1. Both the file system and file classes
2. Both client creation approaches i.e., using `from_connection_string` and using the initializer

I think this is important given the amount of branching logic when it comes to client creation so that there is coverage for the various ways clients are created in `adlfs`